src/QsCompiler/CompilationManager/DataStructures.cs

```diff
@@ -479,9 +479,9 @@ internal struct HeaderEntry<T>
         /// If the symbol of the extracted declaration is not an unqualified symbol,
         /// verifies that it corresponds instead to an invalid symbol and returns null unless the keepInvalid parameter has been set to a string value.
         /// If the keepInvalid parameter has been set to a (non-null) string, uses that string as the SymbolName for the returned HeaderEntry instance.
-        /// Throws an ArgumentException if this verification fails as well.
-        /// Throws an ArgumentException if the extracted declaration is Null.
         /// </summary>
+        /// <exception cref="ArgumentException">The symbol of the extracted declaration is not an unqualified symbol and does not instead correspond to an invalid symbol.</exception>
```

Comment:
I think this is a simpler way to say it:

```suggestion
        /// <exception cref="ArgumentException">The symbol of the extracted declaration is not an unqualified or invalid symbol.</exception>
```